### PR TITLE
Refactor EmailCampaignEntity and TargetGroupEntity

### PR DESCRIPTION
**Pull request by juliawegmayr**
## Description

Refactor `email-campaign` and `target-group` entities to factories as in https://github.com/vivid-planet/comet/blob/main/demo/api/src/dam/entities/dam-file.entity.ts to enable generating them in migrations. 

## Example

For testing the generation of the migrations, I added fields to:

1. add "TESTFILTER" to brevo-contact-attributes

Screenshot 2024-12-10 at 10 49 54

2. add "country" in scope 

Screenshot 2024-12-10 at 10 59 59

## Changeset

[x] I have verified if my change requires a changeset

## Related tasks and documents
https://vivid-planet.atlassian.net/browse/COM-1010

### CODE REVIEW

File: .changeset/happy-cups-design.md
Comment (RainbowBunchie) [inline .changeset/happy-cups-design.md:9]: I would add the information that they need to be passed to the module's config

File: demo/api/schema.gql
Comment (RainbowBunchie) [inline demo/api/schema.gql:538]: The test should be removed

File: demo/api/src/email-campaign/email-campaign-content-scope.ts
Comment (RainbowBunchie) [inline demo/api/src/email-campaign/email-campaign-content-scope.ts:22]: We should remove the test again, since it adds no value to the demo.

File: packages/api/src/brevo-module.ts
Comment (RainbowBunchie) [inline packages/api/src/brevo-module.ts:35]: Can be removed

File: packages/api/src/email-campaign/email-campaign.module.ts
Comment (RainbowBunchie) [inline packages/api/src/email-campaign/email-campaign.module.ts:27]: This should be passed from the module's config

File: packages/api/src/brevo-module.ts
Comment (juliawegmayr) [reply packages/api/src/brevo-module.ts]: 38c4b8a2234b2e6ed98ac735a43f32269f5f615c

File: demo/api/schema.gql
Comment (juliawegmayr) [reply demo/api/schema.gql]: 87d88415598aaa5a8682558906b7e37b972850aa

File: demo/api/src/email-campaign/email-campaign-content-scope.ts
Comment (juliawegmayr) [reply demo/api/src/email-campaign/email-campaign-content-scope.ts]: 87d88415598aaa5a8682558906b7e37b972850aa

File: packages/api/src/email-campaign/email-campaign.module.ts
Comment (juliawegmayr) [reply packages/api/src/email-campaign/email-campaign.module.ts]: 470e1724c6a6dd5262ddd1d5df1e61918dfe510e

File: .changeset/happy-cups-design.md
Comment (juliawegmayr) [reply .changeset/happy-cups-design.md:9]: 066b5a59ab0e57f6a1764580089b4a3b689e585d

**Review by RainbowBunchie — APPROVED** [review]
LGTM, please wait for @raphaelblum's review before merging

**Review by raphaelblum — APPROVED** [review]

**Pull request merged**
